### PR TITLE
[CPDLP] - 847 run some tests to check banding and clawbacks work as expected

### PR DESCRIPTION
**Pull request by StupidCodeFactory**
## Ticket and context

Ticket:

### Code quality checks
- [ ] All commit messages are meaningful and true
- [ ] Added enough automated tests

### HTML Checks
- [ ] All new pages have automated accessibility checks
- [ ] All new pages have visual tests via Percy

### Gotchas
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### How can someone see it it review app?
1. Click the link to review app (posted by a `github-actions` bot below)
2. Follow the steps from the ticket.

## External API changes

Does this change anything in our external APIs? If so, did you remember to update the CHANGELOG for Lead Providers? (docs/source/release-notes)

### CODE REVIEW

File: app/services/early_career_teachers/create.rb
Comment (StupidCodeFactory) [inline app/services/early_career_teachers/create.rb:27]: in a later PR we will need to uncomment those jobs. They spam sidekiq and and review app server with jobs that will fail because the transaction has not yet been comnited.
Will reinstate later when m2-release branch is ready to be merged

File: app/services/mentors/create.rb
Comment (StupidCodeFactory) [inline app/services/mentors/create.rb:28]: https://github.com/DFE-Digital/early-careers-framework/pull/1695/files#r799591130

File: app/components/finance/banding_tracker/band_per_declaration_type_row.html.erb
Comment (asmega) [inline app/components/finance/banding_tracker/band_per_declaration_type_row.html.erb:4]: typo on `retrained_one` should be `retained_one`

File: app/jobs/seed_bpn_test_data.rb
Comment (asmega) [inline app/jobs/seed_bpn_test_data.rb:10]: why does this whole block need to be in a transaction?

File: app/services/early_career_teachers/create.rb
Comment (asmega) [reply app/services/early_career_teachers/create.rb:27]: we probably want to uncomment before we merge. if you want to disable these can we not wrap around a feature toggle?

File: app/services/mentors/create.rb
Comment (asmega) [reply app/services/mentors/create.rb:28]: we probably want to uncomment before we merge. if you want to disable these can we not wrap around a feature toggle?

File: config/database.yml
Comment (asmega) [inline config/database.yml:5]: do we need these changes for the merge?

File: lib/finance/banding_tracker/participant_per_declaration_type_per_band_aggregator.rb
Comment (asmega) [inline lib/finance/banding_tracker/participant_per_declaration_type_per_band_aggregator.rb:1]: as an aside i don't think we should be adding any new code to `/lib`. If we just namespace correctly in `app/`, zeitwerk and rails should load everything correctly and we don't need to require things manually

File: app/jobs/seed_bpn_test_data.rb
Comment (StupidCodeFactory) [reply app/jobs/seed_bpn_test_data.rb:10]: much faster

File: terraform/workspace-variables/review.tfvars
Comment (asmega) [inline terraform/workspace-variables/review.tfvars:12]: are we able to revert this before we merge?

File: terraform/workspace-variables/review_app_env.yml
Comment (asmega) [inline terraform/workspace-variables/review_app_env.yml:6]: are we able to revert this before we merge?

File: lib/finance/banding_tracker/participant_per_declaration_type_per_band_aggregator.rb
Comment (StupidCodeFactory) [reply lib/finance/banding_tracker/participant_per_declaration_type_per_band_aggregator.rb:1]: it is a standalone library, requiring file is fine, it's actually a bit more explicit, at least you know where the code came from. Helps defining boundaries in the code. 
I think we should add more code to `lib` so that we start thinking about problems as domain and respect the boundaries.

File: terraform/workspace-variables/review.tfvars
Comment (StupidCodeFactory) [reply terraform/workspace-variables/review.tfvars:12]: will need this to seed BNP data in m2-release, but will clean this up once we've got the ✅

File: terraform/workspace-variables/review_app_env.yml
Comment (StupidCodeFactory) [reply terraform/workspace-variables/review_app_env.yml:6]: https://github.com/DFE-Digital/early-careers-framework/pull/1695#discussion_r800552898

File: app/services/mentors/create.rb
Comment (StupidCodeFactory) [reply app/services/mentors/create.rb:28]: https://github.com/DFE-Digital/early-careers-framework/pull/1695#discussion_r800552898

I'll revert this. We need to fix this, in a separate PR any. But without those I won't be able to load the seeds, as it spams sidekiq

File: app/services/early_career_teachers/create.rb
Comment (StupidCodeFactory) [reply app/services/early_career_teachers/create.rb:27]: https://github.com/DFE-Digital/early-careers-framework/pull/1695#discussion_r800554240

File: spec/services/mentors/create_spec.rb
Comment (asmega) [inline spec/services/mentors/create_spec.rb:69]: I'm not sure I follow. Do we not have the go-ahead? If we merge this in, we'll have introduced behaviours into the app the don't want

Comment (StupidCodeFactory) [reply spec/services/mentors/create_spec.rb:69]: @asmega the base branch for this PR is `m2-release`. I'm going to merge this then seed the app. We do all the testing we want, than before merging this into `develop` I'll remove those.

**Review by asmega — APPROVED** [review]

**Pull request merged**
